Backport of PR #771 to 0.34 : big int parsing (#798)
cometbft/cometbft: docs/core/subscription.md

response, to query transaction results. See [Indexing
transactions](./indexing-transactions.md) for details.

## Query parameter and event type restrictions

While CometBFT imposes no restrictions on the application with regards to the type of 
the event output, there are several restrictions when it comes to querying 
events whose attribute values are numeric. 

- Queries cannot include negative numbers
- If floating points are compared to integers, they are converted to an integer
- Floating point to floating point comparison leads to a loss of precision for very big floating point numbers

> (`10000000000000000000.0` is treated the same as `10000000000000000000.6`) 

Reviewer: ```suggestion
(e.g., `10000000000000000000.0` is treated the same as `10000000000000000000.6`) 
```

> - When floating points do get converted to integers, they are not rounded, the decimal part is simply truncated.

Reviewer: ```suggestion
- When floating points do get converted to integers, they are always rounded down.
```
